include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3123,6 +3123,242 @@ namespace internal
 
 
 
+      template <int dim, int spacedim, bool level_dof_access>
+      static bool
+      p_refine_flag_set(const DoFCellAccessor<DoFHandler<dim, spacedim>,
+                                              level_dof_access> &accessor)
+      {
+        (void)accessor;
+        // ::DoFHandler does not support p-refinement
+        Assert(
+          false,
+          (typename std::decay<decltype(accessor)>::type::ExcInvalidObject()));
```

Comment:
here and below, can you replace this exception with one that actually says why this can't work?